use_cluster.py

```diff
@@ -366,4 +355,411 @@ def return_cluster_dict(q_cluster, q_ids_list):
         else:
             final_clusters[q_cluster[indx]].append(q_ids_list[indx])
 
-    return final_clusters
\ No newline at end of file
+    return final_clusters
+
+
+"""
+> The objective of this function is to generate labels for each cluster based on a custom score developed by Araz, 
+using Normalised Mutual Information (NMI) and Distance from the Cluster Centroid 
+:The inputs required are the embeddings (generated previously from cluster functions), list of questions, question ids & clusters generated 
+: Read about NMI here: https://arxiv.org/pdf/1702.08199.pdf
+: For how Araz developed the custom score/metric, we will release a Research Paper, which would be attached here. 
+"""
+
+def return_cluster_labels_nmi_ngrams_centroid(embeddings,qs_list,q_ids_list,clusters): 
+
+    labelling_corpus = LabellingClusterCorpus(qs_list, nlp, embeddings, q_ids_list, clusters)
+    
+    cluster_objs, lemmatized_qs_list = generate_modified_qs_data(labelling_corpus.documents, clusters, q_ids_list,1)
+    global_keywords = generate_global_keywords(labelling_corpus.documents, 1)
+    vecs = embed(list(global_keywords))
```

Comment:
This is another variable I would rename for clarity - something like "global_keyword_embeddings."

Reply:
Yes agreed! :D